README.md

```diff
@@ -49,15 +118,349 @@ src/
+
+Beyond defining title font sizes, line heights, and weights:
+
+- **H1** tags are reserved for the page’s title and should not be used; that
+  said, if they are used, they will still show up on the front end as H1 tags
+- **H2** tags populate the ride-side page
+  [table of contents](https://developers.stellar.org/docs/enabling-deposit-and-withdrawal/setting-up-test-server/)
+  in Documentation
+
+## Custom Components
+
+Custom (React) components that are being used throughout Documentation and API
+Reference.
+
+### AttributeTable
+
+<img src="./readme-imgs/attribute-table.png" alt="AttributeTable Component" width="500"/>
+
+`<AttributeTable/>` is currently used in API Reference. Nested Table is allowed.
+
+For example,
+[Api Reference > Resources > Operations > Object](https://developers.stellar.org/api/resources/operations/object/)
+
+```
+import { AttributeTable } from "components/AttributeTable";
+
+<AttributeTable>
+
+- ATTRIBUTE
+  - DATA TYPE
+  - DESCRIPTION
+- amount
+  - string
+  - The amount of `selling_asset` that the account making this offer is willing to sell.
+- price
+  - string
+  - How many units of `selling_asset` it takes to get 1 unit of `buying_asset`. A number representing the decimal form of `price_r`.
+- price_r
+  - object
+  - A precise representation of the buy and sell price of the assets on offer.
+    - n
+      - number
+      - The numerator.
+    - d
+      - number
+      - The denominator.
+
+</AttributeTable>
+```
+
+### Alert
+
+<img src="./readme-imgs/alert.png" alt="Alert Component" width="712"/>
+
+`<Alert/>` is currently used in Documentation.
+
+For example,
+[Seeting Up Test Server](https://developers.stellar.org/docs/enabling-deposit-and-withdrawal/setting-up-test-server/)
+
+```
+import { Alert } from "components/Alert";
+
+<Alert>
+
+Note: the testnet is reset every three months, so when building on it, make sure you have a plan to recreate necessary accounts and other data.  For more info, check out the [best practices for using the testnet](../glossary/testnet.mdx#best-practices-for-using-testnet).
+
+</Alert>
+```
+
+### CodeExample
+
+<img src="./readme-imgs/code-example.png" alt="CodeExample Component" width="389"/>
+
+`<CodeExample/>` is currently used in both API Reference and Documentation. It
+is using [Prism syntax highlighting library](https://prismjs.com/).
+
+You can include snippets for more than one language. See an example of including
+a snippet for `curl` and `JavaScript` below.
+
+For example,
+[Seeting Up Test Server](https://developers.stellar.org/docs/enabling-deposit-and-withdrawal/setting-up-test-server/)
+in Documentation and
+[Resources > Transaction > Retrieve a Transaction](https://developers.stellar.org/api/resources/transactions/single/)
+in API Reference.
+
+````
+import { CodeExample } from "components/CodeExample";
+
+<CodeExample title="Example Request">
+
+```curl
+curl "https://horizon.stellar.org/ledgers/27147222/transactions?limit=2"
+```
+
+```JavaScript
```

Comment:
Should this be `js`? I thought the language here would be the key in the CODE_LANGS lookup, not the value

Reply:
Either works